app/src/main/java/com/globant/weatherapp/activities/WeatherInfoActivity.kt

```diff
@@ -21,7 +21,7 @@ class WeatherInfoActivity : AppCompatActivity() {
         val cityId = intent.extras.getSerializable(CITY_ID)
 
         presenter = WeatherPresenter(WeatherModel(), WeatherView(this))
-        presenter.initPresenter()
+        presenter.initPresenter(cityId.toString().toInt())
```

Comment:
is this really necessary? can't you use only the .toInt() ?

Reply:
Refactored! Thanks Mau